tools/hidl/hidlcomp.cpp

```diff
@@ -6608,28 +6602,20 @@ void EspServInfo::write_esp_client()
         outs("\tif(state!=NULL)\n");
         outs("\t\tstate->Link();\n\n");
 
-        
-        outs("#ifdef _WIN32\n");
-        outs("\t_beginthread(espWorkerThread, 0, (void *)(IRpcRequestBinding *)(esprequest));\n");
-        outs("#else\n");
-
-        outs("\tpthread_attr_t attr;\n");
-        outs("\tpthread_attr_init(&attr);\n");
-        outs("\tpthread_attr_setscope(&attr, PTHREAD_SCOPE_SYSTEM);\n");
-        outs("\tpthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED);\n");
-        outs("\tpthread_attr_setstacksize(&attr, 0x10000);\n");
-        outs("\tThreadId threadid;\n");
-        outs("\tint status;\n");
-        outs("\tdo\n");
+        outs("\n");
+        outf("\tclass CClient%sThread: public Thread\n", name_);
         outs("\t{\n");
-        outf("\t\tstatus = pthread_create(&threadid, &attr, CClient%s::espWorkerThread, (void *)(IRpcRequestBinding *)(esprequest));\n", name_);
-        outs("\t} while (0 != status && (errno == EINTR));\n");
-        outs("\tif (status) {\n");
-        outs("\t\tRelease();\n");
-        outs("\t\tthrow makeOsException(errno);\n");
-        outs("\t}\n");
-
-        outs("#endif\n");
+        outs("\tpublic:\n");
+        outs("\t\tOwned<IRpcRequestBinding> m_esprequest;\n");
+        outf("\t\tCClient%sThread(IRpcRequestBinding *_esprequest) { m_esprequest.set(_esprequest); }\n", name_);
```

Comment:
This should really be a setown, and there needs to be a LINK() when espWorkerThread is called.  As it stands it is accidentally working.